lib/elixir/lib/uri.ex

```diff
@@ -411,7 +411,7 @@ defmodule URI do
   def parse(string) when is_binary(string) do
     # From https://tools.ietf.org/html/rfc3986#appendix-B
     regex = Regex.recompile!(~r/^(([a-z][a-z0-9\+\-\.]*):)?(\/\/([^\/?#]*))?([^?#]*)(\?([^#]*))?(#(.*))?/i)
-    parts = nillify(Regex.run(regex, string))
+    parts = normalize_parts(string, Regex.run(regex, string))
```

Comment:
I think a better fix is to remove the traversal in nillify altogether and do this:

```elixir
parts = Regex.run(regex, string)
destructure [_, _, scheme, _, authority, path, _, query, _, fragment], parts
scheme = nillify(scheme)
authority = nillify(authority)
path = nillify(path)
query = nillify(query)

defp nillify(""), do: nil
defp nillify(other), do: other
```

Thoughts?